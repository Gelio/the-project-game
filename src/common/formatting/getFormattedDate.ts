export function getFormattedDate(date: Date): string {
  const year = date.getFullYear().toString();
  const month = `${date.getMonth() + 1}`.padStart(2, '0');
  const day = `${date.getDay() + 1}`.padStart(2, '0');
  const hour = `${date.getHours()}`.padStart(2, '0');
  const minutes = `${date.getMinutes()}`.padStart(2, '0');
  const seconds = `${date.getSeconds()}`.padStart(2, '0');
  const array = [year, month, day, hour, minutes, seconds];

  return array.join('-');
}
