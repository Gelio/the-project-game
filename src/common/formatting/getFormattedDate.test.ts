import { getFormattedDate } from './getFormattedDate';

describe('getFormattedDate', () => {
  it('should match snapshot', () => {
    const date = new Date(2018, 4, 25, 15, 11, 50);

    const formattedDate = getFormattedDate(date);

    expect(formattedDate).toMatchSnapshot();
  });
});
