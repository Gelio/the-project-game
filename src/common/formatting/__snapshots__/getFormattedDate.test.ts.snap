// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getFormattedDate should match snapshot 1`] = `"2018-05-06-15-11-50"`;
